src/app/services/task-category.service.ts

```diff
@@ -11,8 +17,46 @@ export class TaskCategoryService {
   constructor(private http: Http) { }
 
   public getTaskCategories(): Observable<TaskCategory[]> {
+    const taskCategoryAll = this.getAllCategory();
+    const taskCategoriesCustom = this.getCustomTaskCategories();
+    const taskCategoriesDate = this.getDateTaskCategories();
+
+    return Observable.forkJoin(taskCategoryAll, taskCategoriesCustom, taskCategoriesDate)
+      .map(result => {
+        const allCategory = result[0];
+        const customCategories = result[1];
+        const defaultCategories = result[2];
+        return (<TaskCategory[]>allCategory).concat(defaultCategories).concat(customCategories);
+      }
+    );
+  }
+
+  private getDateTaskCategories(): Observable<TaskCategoryDate[]> {
+    const today = moment().format('YYYY-MM-DD');
+    const tomorrow = moment().add(1, 'days').format('YYYY-MM-DD');
+    const week = moment()
+      .hours(23).minutes(59).seconds(59)
+      .add(7, 'days')
+      .format('YYYY-MM-DD HH:mm:ss');
+
+    return Observable.of([
+      new TaskCategoryDate('Today', today, today),
```

Comment:
It will be great if you hide creating of TaskCategoryDate categories. For example, you can use factory.

Reply:
It will be better to make issue for it.